internal/bminventory/inventory_test.go

```diff
@@ -9874,6 +9874,22 @@ var _ = Describe("TestRegisterCluster", func() {
 				Expect(*actual.Platform.Type).To(Equal(models.PlatformTypeBaremetal))
 			})
 
+			It("vsphere platform and UserManagedNetworking=true", func() {
+				mockClusterRegisterSuccess(true)
+				mockAMSSubscription(ctx)
+
+				params := getClusterCreateParams(nil)
+				params.Platform = &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeVsphere)}
+				params.UserManagedNetworking = swag.Bool(true)
+				reply := bm.V2RegisterCluster(ctx, installer.V2RegisterClusterParams{
+					NewClusterParams: params,
+				})
+				Expect(reply).Should(BeAssignableToTypeOf(installer.NewV2RegisterClusterCreated()))
+				actual := reply.(*installer.V2RegisterClusterCreated).Payload
+				Expect(swag.BoolValue(actual.UserManagedNetworking)).To(Equal(true))
+				Expect(*actual.Platform.Type).To(Equal(models.PlatformTypeVsphere))
```

Comment:
dont we have a way to verify final install config?

Reply:
> do we have a UT that sets all the pltaform things and verify that the install config generated as expected?

We had for the most cases, but we didn't had a case where platform is vSphere and UMN enabled

Reply:
and now? is this UT reads the generated install config? seems to me that it's only 'reading' from the DB

Reply:
No, this UT is only checking the result cluster created on the database. 
I think that if we want to test the generated install-config it should be a separate effort. I'm not familiar with UT that register cluster and verify it install-config

Reply:
cool. so lets verify it as well - even if it takes extra time.

i dont think there was an issue with db in the issue that we recently encountered tbh

Reply:
OK, I'll add a task for it
FYI @gamli75